src/bundle-manifest.ts

```diff
@@ -250,7 +253,10 @@ export function generateShellMergeStrategy(
     generateEagerMergeStrategy(shell),
     generateMatchMergeStrategy(
         (b) => b.files.has(shell) ||
-            b.entrypoints.size >= minEntrypoints && !getBundleEntrypoint(b))
+            b.entrypoints.size >= minEntrypoints && !getBundleEntrypoint(b)),
```

Comment:
Rejiggered and added comments to `generateShellMergeStrategy`.  Perhaps that will help?